File: C++/32_palindromic_array.cpp
/*
Idea: Iterating through each element & finding if it's palindrome or not.
*/

#include<iostream>
#include<string.h>
using namespace std;



class Solution {
public:
    int PalinArray(int a[], int n)
    {
        int temp, num;
        for (int i = 0;i < n;++i)
        {
            temp = 0;
            num = a[i];
            while (num > 0)
            {
                temp = temp * 10 + num % 10;
                num /= 10;
            }
            if (temp != a[i])
                return false;
        }
        return true;
    }
};

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        int a[n];
        for (int i = 0; i < n; i++)
            cin >> a[i];
        Solution obj;
        cout << obj.PalinArray(a, n) << endl;
    }
}

File: Python/Array/32_palindromic_array.py
"""
Idea: Iterating through each element & finding if it's palindrome or not.
"""


def PalinArray(arr, n):
    for i in range(n):
        temp = 0
        num = arr[i]
        while num > 0:
            temp = temp * 10 + num % 10
            num //= 10
        if temp != arr[i]:
            return False
    return True


if __name__ == '__main__':
    t = int(input())
    for j in range(t):
        nn = int(input())
        a = list(map(int, input().strip().split()))
        if PalinArray(a, nn):
            print(1)
        else:
            print(0)
